bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderWatcherHandler.java

```diff
@@ -0,0 +1,291 @@
+            executionJob = scheduler.scheduleWithFixedDelay(periodicExecutionRunnable, config.pollInterval + 5,
+                    config.pollInterval + 5, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        executionJob.cancel(true);
+        initJob.cancel(true);
+        if (ftp.isConnected()) {
+            try {
+                ftp.logout();
+                ftp.disconnect();
+            } catch (IOException e) {
+                logger.debug("Error terminating FTP connection: ", e);
+            }
+        }
+    }
+
+    private ArrayList<String> listDirectory(FTPClient ftpClient, String parentDir, String currentDir, int level,
+            boolean recursive) throws IOException {
+        String dirToList = parentDir;
+        Instant fileTimestamp = null;
+        Instant dateNow = Instant.now();
+        ArrayList<String> dirList = new ArrayList<String>();
+
+        if (!currentDir.equals("")) {
+            dirToList += "/" + currentDir;
+        }
+
+        FTPFile[] subFiles = ftpClient.listFiles(dirToList);
+        int reply = ftpClient.getReplyCode();
+
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            logger.debug("FTP Error: {}", ftpClient.getReplyString());
+            return dirList;
+        }
+        if (subFiles != null && subFiles.length > 0) {
+            for (FTPFile aFile : subFiles) {
+                String currentFileName = aFile.getName();
+                if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                    // skip parent directory and directory itself
+                    continue;
+                }
+
+                if (aFile.isDirectory()) {
+                    if (recursive) {
+                        dirList.addAll(listDirectory(ftpClient, dirToList, currentFileName, level + 1, recursive));
+                    }
+                } else {
+                    fileTimestamp = aFile.getTimestamp().toInstant();
+                    long diff = ChronoUnit.HOURS.between(fileTimestamp, dateNow);
+
+                    if (diff < config.diffHours) {
+                        aFile.setName(dirToList + "/" + currentFileName);
+                        dirList.add("ftp:/" + ftpClient.getRemoteAddress().toString() + aFile.getName());
+                    }
+                }
+            }
+        }
+        return dirList;
+    }
+
+    protected Runnable periodicConnectRunnable = new Runnable() {
+        int reply;
+
+        @Override
+        public void run() {
+            if (ftp == null || ftp.isConnected() == false) {
+                if (config.connectionTimeout <= 0) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Connetction timeout can't be negative");
```

Comment:
```suggestion
                            "Connection timeout can't be negative");
```

You could check the config before creating the connect job. This way you only have to check once.